pnpbridge/src/adapters/src/mqtt_pnp/json_rpc_protocol_handler.cpp

```diff
@@ -87,53 +95,85 @@ JsonRpcProtocolHandler::Initialize(
     }
 }
 
-void
-JsonRpcProtocolHandler::OnPnpMethodCall(
-    const DIGITALTWIN_CLIENT_COMMAND_REQUEST*   CommandRequest,
-    DIGITALTWIN_CLIENT_COMMAND_RESPONSE*        CommandResponse
-)
+void JsonRpcProtocolHandler::OnPnpPropertyCallback(
+    _In_ PNPBRIDGE_COMPONENT_HANDLE /* PnpComponentHandle */,
+    _In_ const char* /* PropertyName */,
+    _In_ JSON_Value* /* PropertyValue */,
+    _In_ int /* version */,
+    _In_ void* /* userContextCallback */)
 {
+    // no-op
+}
+
+int JsonRpcProtocolHandler::OnPnpCommandCallback(
+    _In_ PNPBRIDGE_COMPONENT_HANDLE PnpComponentHandle,
+    _In_ const char* CommandName,
+    _In_ JSON_Value* CommandValue,
+    _Out_ unsigned char** CommandResponse,
+    _Out_ size_t* CommandResponseSize)
+{
+    int result = PNP_STATUS_SUCCESS;
     const char* json_method = nullptr;
     const char* tx_topic = nullptr;
-    JSON_Value *commandParams = nullptr;
+    const char* commandValueString = nullptr;
+
+    MqttPnpInstance* pnpComponent = static_cast<MqttPnpInstance*>(PnpComponentHandleGetContext(PnpComponentHandle));
+    if (pnpComponent != NULL)
+    {
+        if ((commandValueString = json_value_get_string(CommandValue)) == NULL)
+        {
+            LogError("Component %s: Cannot retrieve JSON string for command", pnpComponent->s_ComponentName.c_str());
+            result = PNP_STATUS_BAD_FORMAT;
+        }
 
-    printf("Incoming call to %s with %s\n", CommandRequest->commandName, CommandRequest->requestData);
+        printf("Incoming call to %s with %s\n", CommandName, commandValueString);
 
-    auto iterator = s_Commands.find(CommandRequest->commandName);
-    if (iterator != s_Commands.end()) {
-        json_method = iterator->second.first.c_str();
-        tx_topic = iterator->second.second.c_str();
-    }
+        std::map<std::string, std::pair<std::string, std::string>> commandMap = static_cast<JsonRpcProtocolHandler*> 
+             (pnpComponent->s_ProtocolHandler)->GetCommands();

```

Comment:
>GetCommands [](start = 48, length = 11)

setters/getters unnecessary if this becomes a method of MqttProtocolHandler and we're working on the same instance